service/controller/resource/masters/create_cluster_upgrade_requirement_check.go

```diff
@@ -23,7 +23,15 @@ func (r *Resource) clusterUpgradeRequirementCheckTransition(ctx context.Context,
 
 	isCreating := r.isClusterCreating(cr)
 	anyOldNodes, err := r.anyNodesOutOfDate(ctx)
-	if err != nil {
+	if IsClientNotFound(err) {
+		// The kubernetes API is down.
+		// We check if the Legacy Master VMSS exists and in that case
+		// we assume this is because we're migrating to Flatcar.
+		exists, err := r.vmssExists(ctx, key.ResourceGroupName(cr), key.LegacyMasterVMSSName(cr))
+		if err != nil || !exists {
+			return "", microerror.Mask(err)
```

Comment:
> this return "" will take state machine back to beginning and it gets stuck

Will it take it back? We are not saving the state if the error is returned, right?
https://github.com/giantswarm/azure-operator/blob/e57695f86b29d2488d7e1bdfce8c490c0458f025/service/controller/resource/masters/create.go#L48

Reply:
Not sure what to do when `! exists` is true. That means that the old VMSS is not existing but the new one is either not existing or not working/responding.
We could wait for it to come up (stay in the same state) but that could lead to an infinite loop (if the new VMSS does not exist at all it will never come up).
How do you suggest to handle this error?

Reply:
> We could wait for it to come up (stay in the same state) but that could lead to an infinite loop

By returning the error, I believe that we are staying in the same state, because on error we do not save the returned state (`""` in this case).

Reply:
> By returning the error, I believe that we are staying in the same state, because on error we do not save the returned state

Yes. On error there's no transitioning.